services/download/src/main/java/app/coronawarn/server/services/download/validation/ValidFederationKeyFilter.java

```diff
@@ -1,37 +1,123 @@
 package app.coronawarn.server.services.download.validation;
 
 import app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey;
+import app.coronawarn.server.common.protocols.external.exposurenotification.ReportType;
+import app.coronawarn.server.services.download.DownloadServiceConfig;
+import app.coronawarn.server.services.download.DownloadServiceConfig.Validation;
+import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 /**
- * Responsible for checking fields of the {@link DiagnosisKey} objects contained within batches
- * downloaded from the EFGS. This check is prior to the one executed when building the domain
- * {@link app.coronawarn.server.common.persistence.domain.DiagnosisKey} entity which ensures our
- * data model constraints are not violated for any incoming data channel.
+ * Responsible for checking fields of the {@link DiagnosisKey} objects contained within batches downloaded from the
+ * EFGS. This check is prior to the one executed when building the domain
+ * {@link app.coronawarn.server.common.persistence.domain.DiagnosisKey}
+ * entity which ensures our data model constraints are not violated for any incoming data channel.
  */
 @Component
 public class ValidFederationKeyFilter {
 
   private static final Logger logger = LoggerFactory.getLogger(ValidFederationKeyFilter.class);
+  private final int keyLength;
+  private final List<ReportType> allowedReportTypes;
+  private final int minDsos;
+  private final int maxDsos;
+  private final int minRollingPeriod;
+  private final int maxRollingPeriod;
+  private final int minTrl;
+  private final int maxTrl;
+
+  /**
+   * Constructor for this class.
+   *
+   * @param downloadServiceConfig A {@link DownloadServiceConfig} object.
+   */
+  public ValidFederationKeyFilter(DownloadServiceConfig downloadServiceConfig) {
+    Validation validation = downloadServiceConfig.getValidation();
+    this.keyLength = validation.getKeyLength();
+    this.allowedReportTypes = validation.getAllowedReportTypes();
+    this.minDsos = validation.getMinDsos();
+    this.maxDsos = validation.getMaxDsos();
+    this.minRollingPeriod = validation.getMinRollingPeriod();
+    this.maxRollingPeriod = validation.getMaxRollingPeriod();
+    this.minTrl = validation.getMinTrl();
+    this.maxTrl = validation.getMaxTrl();
+  }
 
   /**
    * Accepts or rejects a key based on the evaluation of the fields against permitted values.
    */
   public boolean isValid(
       app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey federationKey) {
```

Comment:
```suggestion
      DiagnosisKey federationKey) {
```
No longer needed as the `exposurenotification.DiagnosisKey` is the only imported class with `DiagnosisKey` name 🙂